plugins/woocommerce/src/Internal/TransientFiles/TransientFilesEngine.php

```diff
@@ -0,0 +1,496 @@
+		$this->legacy_proxy = $legacy_proxy;
+	}
+
+	/**
+	 * Get the base directory where transient files are stored.
+	 *
+	 * The default base directory is the WordPress uploads directory plus "woocommerce_transient_files". This can
+	 * be changed by using the woocommerce_transient_files_directory filter.
+	 *
+	 * The actual directory for each existing file will be the base directory plus the expiration date
+	 * of the file formatted as 'yyyy-mm-dd'.
+	 *
+	 * @return string Effective base directory where transient files are stored.
+	 * @throws Exception The base directory (possibly changed via filter) doesn't exist.
+	 */
+	public function get_transient_files_directory(): string {
+		$upload_dir_info           = $this->legacy_proxy->call_function( 'wp_upload_dir' );
+		$transient_files_directory = untrailingslashit( $upload_dir_info['basedir'] ) . '/woocommerce_transient_files';
+
+		/**
+		 * Filters the directory where transient files are stored.
+		 *
+		 * Note that this is used for both creating new files (with create_file_by_rendering_template)
+		 * and retrieving existing files (with get_file_by_*).
+		 *
+		 * @param string $transient_files_directory The default directory for transient files.
+		 * @return string The actual directory to use for storing transient files.
+		 *
+		 * @since 8.5.0
+		 */
+		$transient_files_directory = apply_filters( 'woocommerce_transient_files_directory', $transient_files_directory );
+
+		$realpathed_transient_files_directory = $this->legacy_proxy->call_function( 'realpath', $transient_files_directory );
+		if ( false === $realpathed_transient_files_directory ) {
+			throw new Exception( "The base transient files directory doesn't exist: $transient_files_directory" );
+		}
+
+		return untrailingslashit( $realpathed_transient_files_directory );
+	}
+
+	/**
+	 * Create a transient file.
+	 *
+	 * @param string     $file_contents The contents of the file.
+	 * @param string|int $expiration_date A string representing the expiration date formatted as "yyyy-mm-dd", or a number representing the expiration date as a timestamp (the time of day part will be ignored).
+	 * @return string The name of the transient file created (without path information).
+	 * @throws \InvalidArgumentException Invalid expiration date (wrongly formatted, or it's a date in the past).
+	 * @throws \Exception The directory to store the file doesn't exist and can't be created.
+	 */
+	public function create_transient_file( string $file_contents, $expiration_date ): string {
+		if ( is_numeric( $expiration_date ) ) {
+			$expiration_date = gmdate( 'Y-m-d', $expiration_date );
+		} elseif ( ! is_string( $expiration_date ) || ! TimeUtil::is_valid_date( $expiration_date, 'Y-m-d' ) ) {
+			$expiration_date = is_scalar( $expiration_date ) ? $expiration_date : gettype( $expiration_date );
+			throw new InvalidArgumentException( "$expiration_date is not a valid date, expected format: year-month-day" );
```

Comment:
```suggestion
			throw new InvalidArgumentException( "$expiration_date is not a valid date, expected format: YYYY-MM-DD" );
```

This seems a bit less ambiguous about the expected format.